lib/livebook/evaluator/io_proxy.ex

```diff
@@ -249,6 +249,28 @@ defmodule Livebook.Evaluator.IOProxy do
     end
   end
 
+  defp get_chars(encoding, prompt, state, count) do
+    prompt = :unicode.characters_to_binary(prompt, encoding, state.encoding)
+
+    case get_input(prompt, state) do
+      input when is_binary(input) ->
+        {chars, rest} = chars_from_input(input, count)
```

Comment:
We want to pass `encoding` to the function (that's the encoding the user requests) and return chars based on that. If they request `:latin1` then we interpret a every byte as a character, when they request `:unicode` we then count a number of bytes as a single character according to utf8. You can just copy paste the linked code and it should do :)

With that we also don't need the `:unicode.characters_to_binary`.

Reply:
You want this change to be made in the `chars_part` function, because in it I force it as utf8:
`chars_part(<<head::utf8, tail::binary>>, count)`. 

The encoding you want to pass to what function ?

Reply:
What I mean is that I'd do

```elixir
{chars, rest} = chars_from_input(input, encoding, count)
```

Where `chars_from_input` is essentially the function linked before https://github.com/elixir-lang/elixir/blob/d9028a8d0405d19076418376ab50bdc83ff72974/lib/elixir/lib/string_io.ex#L308-L338